crates/next-custom-transforms/tests/fixture/edge-assert/guarded-nodejs/input.js

```diff
@@ -0,0 +1,3 @@
+if (typeof clearImmediate) {
```

Comment:
This doesn't really make sense to write. We should rather test that both positive and negative assertions will ensure the API is not polyfilled<sup>1</sup> i.e.

```tsx
const scheduleTimeoutA = typeof setImmediate === 'function' ? setImmediate : setTimeout
const scheduleTimeoutB = typeof setImmediate !== 'function' ? setTimeout : setImmediate
```

<sup>1</sup> What even was the behavior before this change? That's not explained in the PR description.

Reply:
Previously the linter catches all usages of `process.X` (except env) or node.js specific APIs, even in the case those are properly guarded by typeofs.

Reply:
How did it affect the output of these fixtures?

Reply:
There are also snapshot files generated for lints/errors: https://github.com/vercel/next.js/blob/canary/crates/next-custom-transforms/tests/fixture/edge-assert/process-api/output.stderr
So that fixture also tests that there is no lint/error.